### ISSUE TITLE
pull-request should create the remote branch if it doesn't already exist

### ISSUE THREAD

**Issue by jfirebaugh** [github/hub#165]
When the remote branch on your own repository doesn't exist, the current behavior of pull-request is confusing: it opens the editor with a stub message, and then if you fill out a message and continue, the submission fails with an opaque error message: "Error creating pull request: Unprocessable Entity (HTTP 422)".

A better error message like would help, but in this case, I think pull-request should just automatically push to a new remote branch before opening the pull request, since that's almost always what you want.

**Comment by cabello** [comment]
Since issuing two commands might break people workflow, how about just improve the output a little bit?
```
$ hub pull-request
Error creating pull request: Unprocessable Entity (HTTP 422)
Missing field: "head_sha"
Missing field: "base_sha"
No commits between foo:master and foo:foobar
Did you forget to push the branch?

**Comment by silasdavis** [comment]
yes the `Missing field "head_sha"` etc is pretty unhelpful internal leakage in this case where it might be fairly obvious what has gone wrong

**Comment by gaboesquivel** [comment]
I'm getting this error when trying to git pull-request from a repo that's on an organization I own, repo is not on my account. Consider this use case
